test/test.js

```diff
@@ -1,27 +1,15 @@
 function readBlobAsText(blob) {
   return new Promise(function(resolve, reject) {
-    var reader = new FileReader()
-    reader.onload = function() {
-      resolve(reader.result)
-    }
-    reader.onerror = function() {
-      reject(reader.error)
-    }
-    reader.readAsText(blob)
+    var reader = new FileReaderSync()
```

Comment:
hmm.. gecko has had it for a while now. oh right, i completely forgot this code is shared by window :/